instrumentation/net/http/otelhttp/config.go

```diff
@@ -108,7 +109,16 @@ func WithMeterProvider(provider metric.MeterProvider) Option {
 func WithPublicEndpoint() Option {
 	return optionFunc(func(c *config) {
 		c.PublicEndpoint = true
-		c.SpanStartOptions = append(c.SpanStartOptions, trace.WithNewRoot())
+	})
+}
+
+// WithPublicEndpointFn runs with every request, and allows conditionnally
+// configuring the Handler to link the span with an incoming span context. If
+// this option is not provided or returns false, then the association is a
+// child association instead of a link.
```

Comment:
We should note that WithPublicEndpoint takes precedence over WithPublicEndpointFn